client/src/periodUtils.js

```diff
@@ -42,16 +42,45 @@ export const PERIOD_FACTORIES = {
       .subtract(2 * offset, "weeks")
       .endOf("week")
   }),
-  [RECURRENCE_TYPE.SEMIMONTHLY]: (date, offset) => ({
-    start:
-      date.date() < 15
-        ? date.clone().subtract(offset, "months").startOf("month")
-        : date.clone().subtract(offset, "months").startOf("month").date(15),
-    end:
-      date.date() < 15
-        ? date.clone().subtract(offset, "months").startOf("month").date(14)
-        : date.clone().subtract(offset, "months").endOf("month")
-  }),
+  // for more context read: https://github.com/NCI-Agency/anet/pull/3272#discussion_r515826676
+  [RECURRENCE_TYPE.SEMIMONTHLY]: (date, offset) => {
+    const isCurrentPeriodFirstHalf = date.date() < 15
+    let isTargetPeriodInFirstHalf
+    let targetPeriodMonthStart
+
+    if (offset % 2 === 0) {
+      targetPeriodMonthStart = date
+        .clone()
+        .subtract(offset / 2, "months")
+        .startOf("month")
+      // even number offset means same half with current period
+      isTargetPeriodInFirstHalf = isCurrentPeriodFirstHalf
+    } else {
+      // if odd, things are more complicated
+      // We can't just subtract 15 days due to corner cases like 28 day Feb, 31 day months etc
+      const diffInMonths = isCurrentPeriodFirstHalf
+        ? Math.ceil(offset / 2)
+        : Math.floor(offset / 2)
+
+      targetPeriodMonthStart = date
+        .clone()
+        .subtract(diffInMonths, "months")
+        .startOf("month")
+
+      // since offset is odd, opposite of the date
+      isTargetPeriodInFirstHalf = !isCurrentPeriodFirstHalf
+    }
+
+    return isTargetPeriodInFirstHalf
+      ? {
+        start: targetPeriodMonthStart,
+        end: targetPeriodMonthStart.clone().endOf("day").add(13, "days") // end of day 14
```

Comment:
I have a slight preference for first adding periods, then setting end-of-period, so:
```
        end: targetPeriodMonthStart.clone().add(13, "days").endOf("day") // end of day 14
```